### PR TITLE
Issue 723: Add MiAIRR compliance flag

### PR DESCRIPTION
**Pull request by jday1**
This PR implements the first step for https://github.com/airr-community/airr-standards/issues/723.

A further benefit of this PR is that it fully aligns SyntenyBio's fork with airr-community which will allow us to the airr package directly in the future. This has been ongoing work at Synteny for several months and with this merge and release it will allow us to more actively contribute to AIRR.

### CODE REVIEW

**Comment by javh** [comment]
Thanks!  Given that we are in feature freeze to get the v1.5 release out, this might not make it into v1.5.  I'll take a closer look to be sure.

If not, there will be a lot of merges after the release and we can work this into that.

**Comment by javh** [comment]
From the call:
* No objection to the additional validation level.
* Check if this is the best implementation approach. Eg, instead change the behavior of `validate` to multiple levels (`none`, `miairr`, `strict`, whatever).

File: lang/python/airr/schema.py [hunk 1]
```diff
@@ -42,7 +44,7 @@ class Schema:
     _to_bool_map.update({x: False for x in false_values + [0, False]})
     _from_bool_map = {k: 'T' if v else 'F' for k, v in _to_bool_map.items()}
 
-    def __init__(self, definition):
+    def __init__(self, definition: Union[str, dict]):
```
Comment (scharch) [inline lang/python/airr/schema.py:47]: What is this change doing?

File: lang/python/airr/schema.py [hunk 2]
```diff
@@ -335,15 +338,19 @@ def validate_object(self, obj, missing=True, nonairr=True, context=None):
         # first warn about non-AIRR fields
         if nonairr:
             for f in obj:
-                if context is None: full_field = f
-                else: full_field = context + '.' + f
+                if context is None:
+                    full_field = f
+                else:
+                    full_field = context + '.' + f
```
Comment (scharch) [inline lang/python/airr/schema.py:344]: @javh @bussec what are the guidelines (if any) for style changes like this?

**Comment by jday1** [comment]
I realised this was a duplicate of https://github.com/airr-community/airr-standards/pull/750. I've closed this PR in favour of the other as we have made more progress on that.

**Pull request closed without merging**